tst/CTA.WebForms2Blazor.Tests/Services/WorkspaceManagerServiceTests.cs

```diff
@@ -188,15 +198,94 @@ public void WaitUntilAllDocumentsInWorkspace_Task_Reflects_Cancellations()
             var task = _workspaceBuilder.WaitUntilAllDocumentsInWorkspace(token);
             Assert.False(task.IsCompleted);
 
-            _workspaceBuilder.CreateProjectFile("TestProjectName1");
+            var pid1 = _workspaceBuilder.CreateProjectFile("TestProjectName1");
             Assert.False(task.IsCompleted);
 
-            _workspaceBuilder.AddDocument("TestProjectName1", "TestDocumentName1", "");
+            _workspaceBuilder.AddDocument(pid1, "TestDocumentName1", "");
             Assert.False(task.IsCompleted);
 
             source.Cancel();
             Assert.ThrowsAsync(typeof(OperationCanceledException), async () => await task);
             Assert.True(task.IsCanceled);
         }
+
+        [Test]
+        public async Task GetCurrentDocumentSyntaxTree_Correctly_Builds_Syntax_Tree()
+        {
+            _workspaceBuilder.CreateSolutionFile();
+            var pid1 = _workspaceBuilder.CreateProjectFile("TestProjectName1");
+            var did1 = _workspaceBuilder.AddDocument(pid1, "TestDocumentName1", TestDocument1Text);
+
+            var syntaxTree = await _workspaceBuilder.GetCurrentDocumentSyntaxTree(did1);
+
+            // Generated from TestDocument1Text using Roslyn Quoter site
+            // http://roslynquoter.azurewebsites.net/
+            var targetSyntaxTree = SyntaxFactory.CompilationUnit()
+                .WithMembers(
+                    SyntaxFactory.SingletonList<MemberDeclarationSyntax>(
+                        SyntaxFactory.NamespaceDeclaration(
+                            SyntaxFactory.IdentifierName("TestNamespace1"))
+                        .WithMembers(
+                            SyntaxFactory.SingletonList<MemberDeclarationSyntax>(
+                                SyntaxFactory.ClassDeclaration("TestClass1")
+                                .WithModifiers(
+                                    SyntaxFactory.TokenList(
+                                        SyntaxFactory.Token(SyntaxKind.PublicKeyword)))
+                                .WithMembers(
+                                    SyntaxFactory.SingletonList<MemberDeclarationSyntax>(
+                                        SyntaxFactory.MethodDeclaration(
+                                            SyntaxFactory.PredefinedType(
+                                                SyntaxFactory.Token(SyntaxKind.VoidKeyword)),
+                                            SyntaxFactory.Identifier("TestMethod1"))
+                                        .WithModifiers(
+                                            SyntaxFactory.TokenList(
+                                                SyntaxFactory.Token(SyntaxKind.PublicKeyword)))
+                                        .WithBody(
+                                            SyntaxFactory.Block())))))))
+                .NormalizeWhitespace().SyntaxTree;
+
+            Assert.True(syntaxTree.IsEquivalentTo(targetSyntaxTree));
+        }
```

Comment:
A simpler alternative to comparing syntax trees is to compare the string output. For example:
`Assert.AreEqual(TestDocument1Text, syntaxTree.ToFullString())`

In my opinion, the alternative is easier to read/understand. They are effectively equivalent so feel free to change or keep the same.